### DISCUSSION TITLE
Arm7 w/ Dockerfile

### DISCUSSION THREAD

**Discussion by gitshipdone** [jertel/vuegraf#26] [General]
I do not see any underlying dependency issues, however I am unable to get the docker image to load on Pi4 

Anyone have any luck?

**Comment by jertel** [comment]
Have you tried using `--platform linux/amd64` when running the container? I don't know if it will help, just a guess.

**Reply by gitshipdone** [reply]
Tried this to no avail. I saw some users post they had it working on the Pi (on the emporia forums) however they provided zero explanation of how, going to try to install packages to the dockerfile and see if this is what it is missing. Bloat the image until I can cut it down and do a pull.

**Reply by jertel** [reply]
Do you get some logging output with a clue about the problem?

**Reply by gitshipdone** [reply]
> 
> 
> Do you get some logging output with a clue about the problem?

`standard_init_linux.go:190: exec user process caused "exec format error"`

My guess is its in the alpine image - I'm searching for an arm compatible one will try to build before I start work.

**Reply by gitshipdone** [reply]
Trying arm32v7:3-alpine now

**Reply by gitshipdone** [reply]
Seems like that is the rabbit hole that needs to be investigated to run - If I get the other users dockerfile I will review/clean up and submit a pull when I have some time. While the image did not directly run, now I am getting python errors which is a good start 

> Fatal Python error: init_interp_main: can't initialize time
> Python runtime state: core initialized
> PermissionError: [Errno 1] Operation not permitted
> 
> Current thread 0x76f51390 (most recent call first):
> <no Python frame>

bpendz/vuegrafarm:latest is the image I started working on.

**Reply by jertel** [reply]
I built an arm64 variant at jertel/vuegraf:multiplatform.

![image](https://user-images.githubusercontent.com/16639611/108896542-10835300-75e3-11eb-93e2-88bd035071c1.png)


Give it a shot and see if it helps.

**Reply by gitshipdone** [reply]
> 
> 
> I built an arm64 variant at jertel/vuegraf:multiplatform.
> 
> ![image](https://user-images.githubusercontent.com/16639611/108896542-10835300-75e3-11eb-93e2-88bd035071c1.png)
> 
> Give it a shot and see if it helps.

Unfortunately I am on arm7 (32bit) - thus even with experimental I am getting

`docker: no matching manifest for unknown in the manifest list entries.`

**Reply by jertel** [reply]
There's a linux/arm/v7 variant available now.

The title of this issue references Arm64. Where does that come into the picture with this issue?

**Reply by gitshipdone** [reply]
My apologies on the title it was an oversight. I've since fixed it for SEO. 

Its running now will confirm data on grafana and update thread.

**Reply by gitshipdone** [reply]
On ARM7 after a minute it removes the container and I am seeing: 
```
Fatal Python error: init_interp_main: can't initialize time
Python runtime state: core initialized
PermissionError: [Errno 1] Operation not permitted

Current thread 0x76f8d390 (most recent call first):
<no Python frame>

```

**Reply by jertel** [reply]
Check out this thread:

https://github.com/linuxserver/docker-papermerge/issues/4

Looks like the host needs an updated libseccomp2 to solve this.

**Reply by gitshipdone** [reply]
I can get this to work on arm64 but not arm32bit even with the changes made, I reviewed that doc and im getting permission errors at the python level.

**Reply by gitshipdone** [reply]
It appears this points in the container to the python calls that maybe the issue here https://bugzilla.redhat.com/show_bug.cgi?id=1887445

**Reply by Cjkeenan** [reply]
> On ARM7 after a minute it removes the container and I am seeing:
> 
> ```
> Fatal Python error: init_interp_main: can't initialize time
> Python runtime state: core initialized
> PermissionError: [Errno 1] Operation not permitted
> 
> Current thread 0x76f8d390 (most recent call first):
> <no Python frame>
> ```

Has anyone found a fix for this error yet? I am trying to run this on my RPi 3B and the container keeps throwing this exact error. My `libseccomp2` version is 2.3.3-4 and running as a host process outside of docker runs fine.

Also does the grafana and influxdb instances have to be in docker or can they just be installed as a normal program? I am in the process of migrating everything over to docker, but I wanted to get this working first.

**Reply by jertel** [reply]
No they don't need to be in docker. You can run everything on the host directly.

I don't have an ARM env setup right now so I can't assist on the Python error. But there are ARMv7 images on Docker Hub if you haven't seen them. I'm not sure if that variant will help your situation though.

**Reply by Cjkeenan** [reply]
I am just using :latest via portainer. Should I be doing something other than that or are you saying that there are other docker images specifically for ARM devices?

**Reply by jertel** [reply]
It's possible that your linux machine pulled the correct arch already. You can check it via this command:

```
docker inspect jertel/vuegraf | grep Architecture
```

**Reply by Cjkeenan** [reply]
I think it did pull correctly.
```
pi@BH-3B:~ $ docker inspect jertel/vuegraf | grep Architecture
        "Architecture": "arm",
```

**Comment by jertel** [comment]
FYI - I've removed the temporary `multiplaform` tag, mentioned earlier in this discussion, from DockerHub.  As of today the automated builds will include Arm64 and Arm/v7 variants. This is available now, under the `latest` tag.

**Reply by mabrowning** [reply]
I don't see these images on the `latest` tag. The only digest is `sha256:5b8ebca1358b85d65e3e383128e50f18080f59733a1b786969acbb045ccd3a84` which is for x86_64 it appears. Trying to run on an arm7 32bit Raspberry Pi 3.

**Reply by mabrowning** [reply]
However, `docker build` does produce a compatible image in ~10 minutes and let me skirt the raspbian python dependency problems... Thanks!

**Reply by jertel** [reply]
I've re-enabled the ARM image variants, so those are now available again. I had to disable them earlier this year due to an upstream python/alpine conflict. Thanks for the reminder.
